server/service/query_structure.rs

```diff
@@ -0,0 +1,115 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::collections::HashMap;
+
+use answer::variable::Variable;
+use compiler::executable::pipeline::QueryStructure;
+use ir::pattern::{constraint::Constraint, Vertex};
+use itertools::Itertools;
+
+const JSON_NONE: &str = "{}";
```

Comment:
Is there no way we can encode direcrtly into a JSON type or a Map -> Serialize? 
Is this temporary to enable your testing and this will be refined later?

Reply:
Yes. It's already gone in my branch on http.